javaslang/src/main/java/javaslang/collection/List.java

```diff
@@ -557,13 +557,19 @@
 
     @Override
     default List<T> append(T element) {
-        return foldRight(of(element), (x, xs) -> xs.prepend(x));
+        final ArrayList<T> results = new ArrayList<>(size() + 1);
```

Comment:
Sure, though I need the benchmarks to be merged first, otherwise I don't know what to optimize.
I keep adding commits to this, because they depend on others that aren't merged yet :).